### ISSUE TITLE
Decimal number accuracy isn't automatically provided

### ISSUE THREAD

**Issue by jamespeterschinner** [jamespeterschinner/async_v20#15]
**Description of issue:**

When submitting a new Market Order. OANDA Rejects order due to incorrect decimal number accuracy.
`"rejectReason":"TAKE_PROFIT_ON_FILL_PRICE_PRECISION_EXCEEDED"`

**Current behavior**

async_v20 doesn't round numbers to the instruments accuracy. It currently rounds all decimal numbers to 5 decimal places regardless of what the instrument defines

**Expected behavior**

async_v20 should ensure that all decimal numbers sent to OANDA comply with the instruments accuracy

**Comment by jamespeterschinner** [comment]
I have added a new helper function `_in_context` to format an `OrderRequest` per an instruments specification. And also added a corresponding argument to `OandaClient` to control the behavior of this formatting. See [documentation](http://async-v20.readthedocs.io/en/latest/order_request_formatting.html)

In order to support this. The initialization step now also retrieves all available instruments for an account. (More reason to to initialize first)

The improvement has made an additional class `Unit` that I defined for the purpose of managing decimal number precision redundant. And has been removed. Admittedly I previously failed to understand the relationship between `Instruments` and `OrderRequests`. 

Although this addition doesn't strictly break the API. I feel there may be nuanced behavior implications I cannot predict. (Maybe someone was relying on the rounding of all decimal numbers to 5 places for other purposes). I believe this is also a massive improvement to the library. I will be bumping the major version from 2 to 3 for these reasons
